primefaces/src/main/java/org/primefaces/util/ComponentUtils.java

```diff
@@ -710,6 +724,35 @@ public static <T> T executeInRequestScope(FacesContext context, String var, Obje
         }
     }
 
+    /**
+     * Execute a callback in a stateless scope by temporarily removing and restoring a variable from the FacesContext attributes.
+     *
+     * @param context The FacesContext used for accessing attributes
+     * @param var The variable to be temporarily removed and restored in the context attributes
+     * @param runnable The callback to be executed
+     * @param <E> The type of Throwable that may be thrown
+     * @throws E If the callback throws an exception of type E
+     */
+    public static <E extends Throwable> void executeInStatelessScope(FacesContext context, String var, FailableRunnable<E> runnable) throws E {
+        // if no var passed just execute the callback
+        if (LangUtils.isBlank(var)) {
+            runnable.run();
+            return;
+        }
+
+        Map<Object, Object> stateMap = context.getAttributes();
+
+        Object oldValue = stateMap.remove(var);
+        try {
+            runnable.run();
+        }
+        finally {
+            if (oldValue != null) {
+                stateMap.put(var, oldValue);
+            }
+        }
```

Comment:
same here

Reply:
Ok got it, but I think the name doesn't say much about the method really does

Reply:
Yeah I couldn't think of a better name so I am open to suggestions.  Its basically removing a state variable, running, and restoring a state variable.

Reply:
We can stick with this name for now

Reply:
I have a preference to call it "runInStatelessScope" to make it less verbose, same as the other one